### PR TITLE
surface selection from detected surfaces added

### PR DESCRIPTION
**Pull request by Tanveerbutt**
Surface selection from detected surfaces added
User scan for surfaces and then select one of the surface if there are multiple.
Map will be shown on selected surface.

### CODE REVIEW

File: Assets/WRLD SDK Samples/Materials/Plane Materials/Border Line.mat [hunk 1]
```diff
@@ -0,0 +1,79 @@
+%YAML 1.1
```
Comment (Tanveerbutt) [inline Assets/WRLD SDK Samples/Materials/Plane Materials/Border Line.mat]: We don't need this material as we can change this via line renderer

File: Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs [hunk 2]
```diff
@@ -5,20 +5,61 @@
 using UnityEngine.XR.ARSubsystems;
 
 [RequireComponent(typeof(ARPlaneManager))]
+[RequireComponent(typeof(ARRaycastManager))]
 public class WRLDARMapController : MonoBehaviour
 {
     #pragma warning disable 0649
     [SerializeField] private Transform m_wrldMapMask;
     [SerializeField] private WRLDARStreamingCameraHandler m_streamingCameraHandler;
+    [SerializeField] private GameObject[] m_SurfaceStateMsg;
```
Comment (Tanveerbutt) [inline Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs]: Should only field of text and change text only.

File: Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs [hunk 3]
```diff
@@ -5,20 +5,61 @@
...
     #pragma warning restore 0649
 
     private ARPlaneManager m_arPlaneManager;
+    private ARRaycastManager m_RaycastManager;
```
Comment (Tanveerbutt) [inline Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs]: Naming convention

File: Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs [hunk 4]
```diff
@@ -5,20 +5,61 @@
...
     private ARPlane m_currentARPlane = null;
     private Dictionary<TrackableId, ARPlane> m_detectedPlanes;
+    private List<ARRaycastHit> m_hits = new List<ARRaycastHit>();
+    private bool m_isSurfaceSelected = false;
+    private bool m_isSurfaceDetected = false;
```
Comment (Tanveerbutt) [inline Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs]: These can be removed

File: Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs [hunk 5]
```diff
@@ -5,20 +5,61 @@
...
 
     void Awake() {
         m_detectedPlanes = new Dictionary<TrackableId, ARPlane>();
         m_arPlaneManager = gameObject.GetComponent<ARPlaneManager>();
+        m_RaycastManager = GetComponent<ARRaycastManager>();
+    }
+
+    bool TryGetTouchPosition(out Vector2 touchPosition)
+    {
+            if (Input.touchCount > 0)
```
Comment (Tanveerbutt) [inline Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs]: indentation

File: Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs [hunk 6]
```diff
@@ -5,20 +5,61 @@
...
+            if (m_RaycastManager.Raycast(touchPosition, m_hits, TrackableType.PlaneWithinPolygon))
+            {
+                foreach (var plane in m_detectedPlanes.Values)
+                {
+                    if (plane.trackableId == m_hits[0].trackableId)
+                    {
+                        m_isSurfaceSelected = true;
+                        UpdateSurfaceStateMsg(SurfaceState.Selected);
+                        CheckAndSetCurrentPlane(plane);
+                        DisableDetectedPlaneVisuals(plane.trackableId);
```
Comment (Tanveerbutt) [inline Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs:60]: break loop here

File: Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs [hunk 7]
```diff
@@ -106,4 +145,42 @@ void OnEnable()
...
+            }
+            else
+            {
+                plane.gameObject.SetActive(false);
+            }
+        }
+        m_arPlaneManager.planePrefab.GetComponent<ARPlaneMeshVisualizer>().enabled = false;
+        m_arPlaneManager.planePrefab.GetComponent<MeshCollider>().enabled = false;
+        m_arPlaneManager.planePrefab.GetComponent<MeshRenderer>().enabled = false;
+        m_arPlaneManager.planePrefab.GetComponent<LineRenderer>().enabled = false;
```
Comment (Tanveerbutt) [inline Assets/WRLD SDK Samples/Scripts/WRLDARMapController.cs]: Move repeated code into method

**Review by ArslanJafri — CHANGES_REQUESTED** [review]
Changes needed as discussed on call and added in comments by @Tanveerbutt

**Review by ArslanJafri — APPROVED** [review]

**Pull request merged**